app/services/services.py

```
@@ -0,0 +1,81 @@
+        # First time using this API key? -> Register it
+        if not api_key_data:
+            api_key_data = await api_key_service.validate_and_register(api_key)
+
+        return api_key_data
+
+    except RuntimeError as e:
+        # World not found or other validation errors
+        raise HTTPException(status_code=500, detail=str(e)) from e
+    except Exception as e:
+        # Invalid API key or GW2 API errors
+        raise HTTPException(status_code=401, detail=f"Invalid API key: {str(e)}") from e
```

Ruff (0.14.1)

Use explicit conversion flag

Replace with conversion flag

(RUF010)